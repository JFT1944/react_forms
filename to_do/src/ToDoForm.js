// import e from 'express'
import React, {useState} from 'react'
import ToDoList from './TodoList'
import ToDo from './toDo'
// import e from 'express'

function ToDoForm(props){
const [oldValue, newValue] = useState('')
    function updatedText(e){
        newValue(e.target.value)
    }

    function addNewItem(e){
        e.preventDefault()
        console.log(e)
        props.updateTheList(oldValue)

    }

    return(
        <form>
            <input value={oldValue} type="text" onChange={(evt) => updatedText(evt)} data-testid='input_form' ></input>
            <input type="submit" onClick={(evt) => addNewItem(evt)} data-testid='subBut'></input>

        </form>
    )
}

export default ToDoForm