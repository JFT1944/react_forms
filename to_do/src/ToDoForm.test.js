
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import  ToDoForm from "./ToDoForm";
// import TEST_IMAGES from "./_testCommon.js";
import App from './App'



it("renders without crashing", async function() {
    render(<ToDoForm  />);
  });


it("matches snapshot", function() {
    const {asFragment} = render(<ToDoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
  

// Tests to make, then turn in.
// it('should create a new todo', function(){
//     let {getByTestId, getByText} = render(<ToDoForm />)
//     let submitB = getByTestId('subBut')
//     expect(submitB).toBeInTheDocument()
//     fireEvent.click(submitB)
// })

