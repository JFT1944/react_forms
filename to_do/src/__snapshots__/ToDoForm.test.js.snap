// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="input_form"
      type="text"
      value=""
    />
    <input
      data-testid="subBut"
      type="submit"
    />
  </form>
</DocumentFragment>
`;
